programs/pkey/gen_key.c

```diff
@@ -434,15 +435,13 @@ int main( int argc, char *argv[] )
 
 exit:
 
+#if defined(MBEDTLS_ERROR_C)
     if( exit_code != MBEDTLS_EXIT_SUCCESS )
```

Comment:
Please improve these tests as done in: https://github.com/ARMmbed/mbedtls/pull/1572/files#diff-8014a58f3a24d70136af9ebc25c30eb9R444